### PR TITLE
NO-ISSUE reduce main cyclomatic complexity

### PR DESCRIPTION
**Pull request by filanov**

### CODE REVIEW

File: cmd/main.go [hunk 1]
```diff
@@ -246,40 +225,27 @@ func main() {
...
-			log.WithError(err).Fatalf("Failed to create client for OCP")
-		}
+		mustSucceedf(err, "Failed to create client for OCP")
 
 	case deployment_type_onprem, deployment_type_ocp:
 		lead = &leader.DummyElector{}
 		autoMigrationLeader = lead
 		// in on-prem mode, setup file system s3 driver and use localjob implementation
 		objectHandler = s3wrapper.NewFSClient("/data", log)
-		if objectHandler == nil {
```
Comment (YuviGold) [inline cmd/main.go:264]: Was it removed on purpose?

File: cmd/main.go [hunk 2]
```diff
@@ -298,12 +264,6 @@ func main() {
 	hostStateMonitor.Start()
 	defer hostStateMonitor.Stop()
 
-	if newUrl, err = s3wrapper.FixEndpointURL(Options.BMConfig.S3EndpointURL); err != nil {
```
Comment (YuviGold) [inline cmd/main.go:301]: Was it removed on purpose?

File: cmd/main.go (hunk 1 above)
Comment (filanov) [reply cmd/main.go:264]: I looked at the function and it return a status object without any logic.

File: cmd/main.go (hunk 2 above)
Comment (filanov) [reply cmd/main.go:301]: I see the same call here https://github.com/openshift/assisted-service/pull/830/files#diff-c444f711e9191b53952edb65bfd8c644419fc7695c62611dc0fb304b4fb197d6R191 
It wasn't clear to me why it's called twice.

File: cmd/main.go [hunk 3]
```diff
@@ -125,42 +125,32 @@ func main() {
 		log.Fatal(err.Error())
 	}
 
+	mustSucceedf := func(err error, msg string, args ...interface{}) {
```
Comment (slaviered) [inline cmd/main.go:128]: perhaps choose a clearer name such a callOrAbort or abortOnFailure

File: cmd/main.go (hunk 2 above)
Comment (YuviGold) [reply cmd/main.go:301]: That's `Options.JobConfig.S3EndpointURL` and `Options.BMConfig.S3EndpointURL` .

File: cmd/main.go (hunk 3 above)
Comment (filanov) [reply cmd/main.go:128]: it's always call anything, maybe failOnError?

**Review by YuviGold — APPROVED** [review]
LGTM

**Pull request merged**